pkg/runtime/security/auth.go

```
@@ -153,8 +155,28 @@ func (a *authenticator) CreateSignedWorkloadCert(id, namespace, trustDomain stri
 	return signedCert, nil
 }
 
-// currently we support Kubernetes identities.
-func getToken() string {
+// Returns the token for authenticating with Sentry.
+func getToken() (token string, validator sentryv1pb.SignCertificateRequest_TokenValidator) {
+	// Check if we have a token in the DAPR_SENTRY_TOKEN env var (for the JWKS validator)
+	if v := os.Getenv(consts.SentryTokenEnvVar); v != "" {
+		log.Debug("Loaded token from DAPR_SENTRY_TOKEN environment variable")
+		return v, sentryv1pb.SignCertificateRequest_JWKS
+	}
+
+	// Check if we have a token file in the DAPR_SENTRY_TOKEN_FILE env var (for the JWKS validator)
+	if path := os.Getenv(consts.SentryTokenFileEnvVar); path != "" {
+		// Attempt to read the file
+		// Errors are logged but we still return the value even if empty
+		b, err := os.ReadFile(path)
+		if err != nil {
+			log.Warnf("Failed to read token at path '%s': %v", path, err)
```

We _must_ return an error here. Silently ignoring this puts us in a wired state and has will have consequences downstream.

The caller can handle waiting for the file to exist in later PRs.

---

Done

---

@ItalyPaleAle maybe you are missing a push?